pkg/cloud/openstack/machine/actuator.go

```diff
@@ -145,7 +145,16 @@ func (oc *OpenstackClient) Create(ctx context.Context, cluster *clusterv1.Cluste
 		userDataRendered = string(userData)
 	}
 
-	instance, err = machineService.InstanceCreate(fmt.Sprintf("%s/%s", cluster.ObjectMeta.Namespace, cluster.Name), machine.Name, providerSpec, userDataRendered, providerSpec.KeyName)
+	var clusterSpec openstackconfigv1.OpenstackClusterProviderSpec
```

Comment:
there is already a function doing this for us. It also checks `Value` for being nil.

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/pkg/apis/openstackproviderconfig/v1alpha1/register.go#L42

Reply:
ah, good to know. Ill change that